cpp/tests/utilities/column_utilities.cu

```diff
@@ -69,5 +145,81 @@ void expect_equal_buffers(void const* lhs, void const* rhs,
                             typed_rhs));
 }
 
+struct column_view_printer {
+  template <typename Element, typename std::enable_if_t<is_numeric<Element>()>* = nullptr>
+  void operator()(cudf::column_view const& col, std::vector<std::string> & out) {
+    auto h_data = cudf::test::to_host<Element>(col);
+
+    out.resize(col.size());
+
+    if (col.nullable()) {
+      size_type index = 0;
+      std::transform(h_data.first.begin(), h_data.first.end(), out.begin(), [&h_data, &index](Element el) {
+          return (bit_is_set(h_data.second.data(), index++)) ? std::to_string(el) : std::string("@");
+        });
+    } else {
+      std::transform(h_data.first.begin(), h_data.first.end(), out.begin(), [](Element el) {
+          return std::to_string(el);
+        });
+    }
+  }
+
+  template <typename Element, typename std::enable_if_t<is_timestamp<Element>()>* = nullptr>
+  void operator()(cudf::column_view const& col, std::vector<std::string> & out) {
+    //
+    //  For timestamps, convert timestamp column to column of strings, then
+    //  call string version
+    //
+    auto col_as_strings = cudf::strings::from_timestamps(col);
+
+    this->template operator()<cudf::string_view>(*col_as_strings, out);
+  }
+
+  template <typename Element, typename std::enable_if_t<is_compound<Element>()>* = nullptr>
```

Comment:
While strings are the only compound type that currently exist, this should be more precise and only be enabled for a strings column. 
```suggestion
  template <typename Element, typename std::enable_if_t<is_same<Element,string_view>::value()>* = nullptr>
```